tags/tags-api/api/src/java/org/sakaiproject/tags/api/TagCollection.java

```diff
@@ -0,0 +1,182 @@
+        return tagCollectionId;
+    }
+
+    public void setTagCollectionId(String tagId) throws IllegalStateException {
+        if ( this.tagCollectionId != null ) {
+            throw new IllegalStateException("Cannot change a tagCollectionId once set");
+        }
+        this.tagCollectionId = tagId;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public String getCreatedBy() {
+        return createdBy;
+    }
+
+    public void setCreatedBy(String createdBy) {
+        this.createdBy = createdBy;
+    }
+
+    public long getCreationDate() {
+        return creationDate;
+    }
+
+    public void setCreationDate(long creationDate) {
+        this.creationDate = creationDate;
+    }
+
+    public String getExternalSourceName() {
+        return externalSourceName;
+    }
+
+    public void setExternalSourceName(String externalSourceName) {
+        this.externalSourceName = externalSourceName;
+    }
+
+    public String getExternalSourceDescription() {
+        return externalSourceDescription;
+    }
+
+    public void setExternalSourceDescription(String externalSourceDescription) {
+        this.externalSourceDescription = externalSourceDescription;
+    }
+
+    public String getLastModifiedBy() {
+        return lastModifiedBy;
+    }
+
+    public void setLastModifiedBy(String lastModifiedBy) {
+        this.lastModifiedBy = lastModifiedBy;
+    }
+
+    public long getLastModificationDate() {
+        return lastModificationDate;
+    }
+
+    public void setLastModificationDate(long lastModificationDate) {
+        this.lastModificationDate = lastModificationDate;
+    }
+
+    public Boolean getExternalCreation() {
+        return externalCreation;
+    }
+
+    public void setExternalCreation(Boolean externalCreation) {
+        this.externalCreation= externalCreation;
+    }
+
+    public Boolean getExternalUpdate() {
+        return externalUpdate;
+    }
+
+    public void setExternalUpdate(Boolean externalUpdate) {
+        this.externalUpdate= externalUpdate;
+    }
+
+    public long getLastSynchronizationDate() {
+        return lastSynchronizationDate;
+    }
+
+    public void setLastSynchronizationDate(long lastSynchronizationDate) {
+        this.lastSynchronizationDate = lastSynchronizationDate;
+    }
+
+    public long getLastUpdateDateInExternalSystem() {
+        return lastUpdateDateInExternalSystem;
+    }
+
+    public void setLastUpdateDateInExternalSystem(long lastUpdateDateInExternalSystem) {
+        this.lastUpdateDateInExternalSystem = lastUpdateDateInExternalSystem;
+    }
+
+    public Errors validate() {
+        Errors errors = new Errors();
+        //Maybe here we can validate the tag Collection values
```

Comment:
Yes, it is the same, finally I didn't required any extra validations, so I can change the message to say something like "In the case new developments require extra validations, here is the place to add them" to help future developers.